root/components/list/AreaList.js

```diff
@@ -29,63 +31,24 @@ const AreaList = ({
   checkboxes,
   order,
   sortable,
-}: Props) => (
-  <table className="tbl">
-    <thead>
-      <tr>
-        {$c.user_exists && checkboxes ? (
-          <th className="checkbox-cell">
-            <input type="checkbox" />
-          </th>
-        ) : null}
-        <th>
-          {sortable
-            ? (
-              <SortableTableHeader
-                label={l('Area')}
-                name="name"
-                order={order}
-              />
-            )
-            : l('Area')}
-        </th>
-        <th>
-          {sortable
-            ? (
-              <SortableTableHeader
-                label={l('Type')}
-                name="type"
-                order={order}
-              />
-            )
-            : l('Type')}
-        </th>
-      </tr>
-    </thead>
-    <tbody>
-      {areas.map((area, index) => (
-        <tr className={loopParity(index)} key={area.id}>
-          {$c.user_exists && checkboxes ? (
-            <td>
-              <input
-                name={checkboxes}
-                type="checkbox"
-                value={area.id}
-              />
-            </td>
-          ) : null}
-          <td>
-            <DescriptiveLink entity={area} />
-          </td>
-          <td>
-            {area.typeName
-              ? lp_attributes(area.typeName, 'area_type')
-              : null}
-          </td>
-        </tr>
-      ))}
-    </tbody>
-  </table>
-);
+}: Props) => {
+  const checkboxColumn = $c.user_exists && checkboxes
+    ? defineCheckboxColumn(checkboxes)
+    : null;
+  const nameColumn =
+    defineNameColumn<CoreEntityT>(l('Area'), order, sortable);
+  const typeColumn = defineTypeColumn('area_type', order, sortable);
+
+  const columns = React.useMemo(
+    () => [
+      ...(checkboxColumn ? [checkboxColumn] : []),
+      nameColumn,
+      typeColumn,
+    ],
+    [],
+  );
```

Comment:
Oh, this kinda defeats the point of `React.useMemo`. The one in root/user/UserCollections.js is wrong too, though I overlooked that.

It's supposed to be for avoiding recalculating things on every render, so to make this right, we should move all the column declarations inside the `useMemo` callback. Then, we should fix the dependencies array (the second argument to `useMemo`) to list all the variables used to create the columns... i.e. `[$c.user_exists, checkboxes, order, sortable]`.

React will then check if any of the values in the dependencies array changed to decide if it needs to recalculate the columns.